src/xenia/app/discord/discord_presence.cc

```diff
@@ -0,0 +1,63 @@
+/**
+******************************************************************************
+* Xenia : Xbox 360 Emulator Research Project                                 *
+******************************************************************************
+* Copyright 2015 Ben Vanik. All rights reserved.                             *
+* Released under the BSD license - see LICENSE in the root for more details. *
+******************************************************************************
+*/
+
+#include "discord_presence.h"
+#include "third_party/discord-rpc/include/discord_rpc.h"
+
+namespace xe {
+namespace discord {
+
+void HandleDiscordReady(const DiscordUser* request) {}
+void HandleDiscordError(int errorCode, const char* message) {}
+void HandleDiscordJoinGame(const char* joinSecret) {}
+void HandleDiscordJoinRequest(const DiscordUser* request) {}
+void HandleDiscordSpectateGame(const char* spectateSecret) {}
+
+void DiscordPresence::InitializeDiscord() {
+  DiscordEventHandlers handlers;
+  memset(&handlers, 0, sizeof(handlers));
+  handlers.ready = &HandleDiscordReady;
+  handlers.errored = &HandleDiscordError;
+  handlers.joinGame = &HandleDiscordJoinGame;
+  handlers.joinRequest = &HandleDiscordJoinRequest;
+  handlers.spectateGame = &HandleDiscordSpectateGame;
+
+  Discord_Initialize("441627211611766784", &handlers, 0, "");
+}
+
+void DiscordPresence::NotPlaying() {
+  DiscordRichPresence discordPresence;
+  memset(&discordPresence, 0, sizeof(discordPresence));
+  discordPresence.state = "Idle";
+  discordPresence.details = "Standby";
+  discordPresence.largeImageKey = "default";
+  discordPresence.instance = 1;
+
+  Discord_UpdatePresence(&discordPresence);
+}
+
+void DiscordPresence::PlayingTitle(std::wstring game_title) {
+  DiscordRichPresence discordPresence;
+  memset(&discordPresence, 0, sizeof(discordPresence));
+  discordPresence.state = "In Game";
+  char mb_game_title[128];  // buffer to hold game title (max 128 bytes)
+  std::wcstombs(mb_game_title, game_title.c_str(),
```

Comment:
This is bad and will convert based on current based on current C locale. Strings provided to Discord rich presence API should be in UTF8.

Should probably use WideCharToMultiByte on Windows because we can explicitly specify the code page there. Not sure about options on posix that don't involve changing C locale.